src/components/Identity/createComponent.js

```diff
@@ -133,8 +133,16 @@ export default (processIdSyncs, config, logger, consent, eventManager) => {
         // and now we have the identity cookie, we can let the queued
         // requests go out. Technically, we should always have an identity
         // cookie at this point, but we check just to be sure.
-        if (deferredForIdentityCookie && hasIdentityCookie()) {
-          deferredForIdentityCookie.resolve();
+        if (deferredForIdentityCookie) {
+          if (hasIdentityCookie()) {
+            deferredForIdentityCookie.resolve();
+          } else {
+            // This logic assumes that the code setting the cookie is working as expected and that
+            // the cookie was missing from the response.
+            const noIdentityCookieErrorMsg = `An identity was not set properly. Please verify that the org ID ${orgId} 
+            configured in Alloy matches the org ID specified in the edge configuration.`;
+            return deferredForIdentityCookie.reject(noIdentityCookieErrorMsg);
```

Comment:
We've been rejecting promises with error instances for consistency.

Reply:
It seems like the promise rejection is always unhandled, even when I try to handle it from sandbox/public/index.html. For example:

```
alloy("event", {
        viewStart: true,
        scopes: ["alloy-location-1", "alloy-location-2"],
        xdm: {
          // Demonstrates overriding automatically collected data
          device: {
            screenHeight: 1
          }
        }
      }).then(function(data) {
        console.log("Sandbox: View start event has completed.", data);
      }).catch(error => {
        console.log(error);
        debugger;
      });
```

We need to figure out why the promise that's handed back to customers isn't being rejected under this scenario and add an e2e test for it. Want my help with this?

Reply:
Yes please @Aaronius  :) If you can make the time.

Reply:
👍 K, I'll hop on it when I'm done with my current task.

Reply:
Merci.